### PR TITLE
[NPUW] Recent coverity fix

### PR DESCRIPTION
**Pull request by smirnov-alexey**

### CODE REVIEW

File: src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp [hunk 1]
```diff
@@ -473,7 +473,6 @@ DQMatMulGQiP::DQMatMulGQiP(Context::Ref ctx) {
         auto qweight_shape = matched_qweight->output(0).get_shape();
         auto qcoeff_shape = matched_qcoeff->output(0).get_shape();
         auto act_shape = matched_out_mmi.get_shape();
-        auto out_shape = matched_node_matmul->output(0).get_shape();
```
Comment (AsyaPronina) [inline src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp:476]: it is not used?

File: src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp [hunk 2]
```diff
@@ -1034,10 +1032,10 @@ HostGather::HostGather(Context::Ref ctx) {
     auto callback = [=](ov::pass::pattern::Matcher& m) {
         auto& node_to_output = m.get_pattern_value_map();
         auto out_shape = node_to_output.at(qgthrw).get_shape();
-        auto matched_out_qweight = node_to_output.at(qweight);
+        auto& matched_out_qweight = node_to_output.at(qweight);
```
Comment (AsyaPronina) [inline src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp:1035]: shouldn't be const, yes?

File: src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp [hunk 3]
```diff
@@ -1173,8 +1171,6 @@ DQUnpackDictMatMulCWu::DQUnpackDictMatMulCWu(Context::Ref ctx) {
         auto matched_result = std::static_pointer_cast<ov::op::v0::Result>(matched_node_res);
 
         auto qcoeff_shape = matched_qcoeff->output(0).get_shape();
-        auto qzerop_shape = matched_qzerop->output(0).get_shape();
-        auto act_shape = matched_mmi.get_shape();
```
Comment (AsyaPronina) [inline src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp:1174]: not used?

File: src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp [hunk 4]
```diff
@@ -1228,8 +1224,7 @@ DQUnpackDictMatMulGQi::DQUnpackDictMatMulGQi(Context::Ref ctx) {
         auto matched_matmul = std::static_pointer_cast<ov::op::v0::MatMul>(matched_node_matmul);
         auto matched_result = std::static_pointer_cast<ov::op::v0::Result>(matched_node_res);
 
-        auto qcoeff_shape = matched_qcoeff->output(0).get_shape();
-        auto act_shape = matched_mmi.get_shape();
```
Comment (AsyaPronina) [inline src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp:1232]: also not used?

File: src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp (hunk 1 above)
Comment (smirnov-alexey) [reply src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp:476]: Yes

File: src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp (hunk 2 above)
Comment (smirnov-alexey) [reply src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp:1035]: Not sure, coverity requested `auto&`

File: src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp (hunk 3 above)
Comment (smirnov-alexey) [reply src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp:1174]: Yes

File: src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp (hunk 4 above)
Comment (smirnov-alexey) [reply src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp:1232]: Yes

**Review by AsyaPronina — APPROVED** [review]

**Review by dmatveev — APPROVED** [review]
No moves - already good :)

**Pull request merged**